app/controllers/users/approvals_controller.rb

```diff
@@ -10,12 +10,13 @@ def new
   end
 
   def create
-    user = User.find_by(email: allowed_params[:approvable].downcase)
-    approval = Approval.add_friend(current_user, user) if user
-    if approval_created?(user, approval)
-      approvals_presenter_and_gon('User')
-      redirect_to(user_friends_path, notice: 'Friend request sent')
-    end
+    result = CreateApproval.call(
```

Comment:
Introduced interactor here in order to get rid of fat methods and complex logic with redirects in private method.